### ISSUE TITLE
Don't call actions in component constructors

### ISSUE THREAD

**Issue by jedireza** [jedireza/aqua#248]
Instead of calling actions to get data in component constructors, we should probably move those calls to `componentDidMount()`.

> #### `componentDidMount()`
> [...]
> If you need to load data from a remote endpoint, this is a good place to instantiate the network request.

Via: https://facebook.github.io/react/docs/react-component.html#componentdidmount

Examples of calling actions in constructors:
 - https://github.com/jedireza/aqua/blob/e80fafe106e46318c851bdbd804cfbeef537a83e/client/pages/admin/accounts/search/index.jsx#L26
 - https://github.com/jedireza/aqua/blob/e80fafe106e46318c851bdbd804cfbeef537a83e/client/pages/admin/accounts/details/index.jsx#L26-L27

**Comment by jaburcodes** [comment]
I can take it,

Is it done in all project components?

Or just a few?

**Comment by jedireza** [comment]
I wouldn't say that it's not the right place for the calls, but probably a better place. I experienced a case where different components were firing actions in their constructor (causing changes in a redux store) and getting state from the same stores caused an invariant of modifying state during a render. After some digging I moved the action calls to `componentDidMount()` and that resolved the issue.

This could also be better for server side rendering because IIRC, the components don't "mount" on the server.
